### PR TITLE
[DeviceProvisioningService] Manage Azure IoT Hub Device Provisioning Service registrations.

### PR DESCRIPTION
**Pull request by anusapan**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by anusapan** [comment]
Design review : https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/603

File: src/DeviceProvisioningServices/DeviceProvisioningServices/Models/PSDeviceRegistrationState.cs [hunk 1]
```diff
@@ -76,4 +76,7 @@ public class PSDeviceRegistrationState
         [JsonProperty(PropertyName = "etag", DefaultValueHandling = DefaultValueHandling.Ignore)]
         public string ETag { get; internal set; }
     }
+
+    public class PSDeviceRegistrationStates : PSDeviceRegistrationState
+    { }
```
Comment (VeryEarly) [inline src/DeviceProvisioningServices/DeviceProvisioningServices/Models/PSDeviceRegistrationState.cs:81]: Why is this PSDeviceRegistrationStates neede?

File: src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            {
+                DeviceRegistrationState result = client.GetDeviceRegistrationStateAsync(this.RegistrationId).GetAwaiter().GetResult();
+                this.WriteObject(IotDpsUtils.ToPSDeviceRegistrationState(result));
+            }
+
+            if (this.IsParameterBound(c => c.EnrollmentId))
+            {
+                string query = this.IsParameterBound(c => c.Query) ? this.Query : "select * from enrollments";
+                QueryResult results = client.CreateEnrollmentGroupRegistrationStateQuery(new QuerySpecification(query), this.EnrollmentId).NextAsync().GetAwaiter().GetResult();
+                this.WriteObject(IotDpsUtils.ToPSDeviceRegistrationStates(results.Items), true);
```
Comment (VeryEarly) [inline src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs:106]: Can you please explain why not return PSDeviceRegistrationState?

File: src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    using Microsoft.Azure.Commands.Management.DeviceProvisioningServices.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Devices.Provisioning.Service;
+    using Microsoft.Azure.Management.DeviceProvisioningServices;
+    using Microsoft.Azure.Management.DeviceProvisioningServices.Models;
+    using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "IoTDeviceProvisioningServiceRegistration", DefaultParameterSetName = ResourceParameterSet)]
+    [Alias("Get-" + ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "IoTDpsRegistration")]
+    [OutputType(typeof(PSDeviceRegistrationState), typeof(PSDeviceRegistrationStates[]))]
```
Comment (VeryEarly) [inline src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs:29]: It looks like it's not necessary to have two output types

File: src/DeviceProvisioningServices/DeviceProvisioningServices/Models/PSDeviceRegistrationState.cs (hunk 1 above)
Comment (anusapan) [reply src/DeviceProvisioningServices/DeviceProvisioningServices/Models/PSDeviceRegistrationState.cs:81]: The command will return either a single object or multiple objects, So having different output types is required to mention in .ps1xml format file to display the output of multiple objects in table format and single object in list format.
`PSDeviceRegistrationStates: Use to display the output in table format`
`PSDeviceRegistrationState: Use to display the output in list format`

File: src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs (hunk 2 above)
Comment (anusapan) [reply src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs:106]: This section will return multiple items of Registration

File: src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs (hunk 3 above)
Comment (anusapan) [reply src/DeviceProvisioningServices/DeviceProvisioningServices/Cmdlet/IotDpsDataPlane/Registration/GetAzIotDeviceProvisioningServiceRegistration.cs:29]: This is only to display the output in different format

**Pull request merged**